flatfs.go

```diff
@@ -756,43 +797,54 @@ func (fs *Datastore) checkpointDiskUsage(newDuInt int64) {
 	}
 }
 
+// persistDiskUsageFile updates the diskusage file with the last known
+// value
 func (fs *Datastore) persistDiskUsageFile() {
-	// Store DiskUsage on clean shutdowns.
-	du, err := fs.DiskUsage()
-	if err != nil {
-		// do not store on errors. just ignore them
-		return
-	}
+	fs.updateLock.Lock()
+	defer fs.updateLock.Unlock()
 
-	atomic.StoreInt64(&fs.diskUsageCheckpoint, int64(du))
+	du := atomic.LoadInt64(&fs.diskUsage)
+
+	origVal := fs.storedValue.diskUsage
+	// update the stored diskUsage value now to prevent unnecessary
+	// calls to persistDiskUsageFile.  On error role back the value to
+	// the original
+	atomic.StoreInt64(&fs.storedValue.diskUsage, du)
 
-	duB := []byte(fmt.Sprintf("%d", du))
 	tmp, err := ioutil.TempFile(fs.path, "du-")
 	if err != nil {
+		atomic.StoreInt64(&fs.storedValue.diskUsage, origVal)
```

Comment:
Should definitely be logging these errros.

Reply:
Okay.